src/components/BlogForm/NewPost.tsx

```diff
@@ -0,0 +1,117 @@
+"use client"; // needed for useform
+import { Button, Stack, TextField, Typography } from "@mui/material";
+import { useForm, Controller } from "react-hook-form";
+import { postBlog } from "../../api/blog";
+import { useSession } from "next-auth/react";
+import { DataProps } from "./types";
+
+const NewPost = () => {
+  const { data: session } = useSession();
+  const user = session?.user.name as string;
+
+  const {
+    control,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({
+    defaultValues: {
+      author: "",
+      title: "",
+      body: "",
+      date: "",
+      tags: "",
```

Comment:
tags plural is string? maybe add a type for the form state

Reply:
i have some ideas for tags in the future but going to do it later. not on my mind atm hence its a string for now

Reply:
ok cool, we'll want to make tags a predetermined set of strings in the future, maybe only able to update them through the admin panel so that there's consistency across blogs and Ken can't just create whatever he wants